src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -47139,6 +47130,1576 @@ void M5(object? x)
+            : x.ToString(); // 1
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_14"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_14()
+        {
+            var source = @"
+class C
+{
+    long? M0(object obj) => null;
+    void M(C? c, object? x, int? i)
+    {
+        _ = (c?.M0(x = 0) == i)
+            ? x.ToString() // 1
+            : x.ToString(); // 2
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (8,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(8, 15),
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_15"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_15()
+        {
+            var source = @"
+class C
+{
+    C M0(object obj) => this;
+    void M(C? c, object? x)
+    {
+        _ = ((object?)c?.M0(x = 0) != null)
+            ? x.ToString()
+            : x.ToString(); // 1
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_16"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_16()
+        {
+            var source = @"
+class C
+{
+    void M(object? x)
+    {
+        _ = ""a""?.Equals(x = 0) == true
+            ? x.ToString()
+            : x.ToString();
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_17"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_17()
+        {
+            var source = @"
+class C
+{
+    void M(C? c, object? x, object? y)
+    {
+        _ = (c?.Equals(x = 0), c?.Equals(y = 0)) == (true, true)
+            ? x.ToString() // 1
+            : y.ToString(); // 2
+    }
+}
+";
+            // This could be made to work (i.e. removing diagnostic 1) but isn't a high priority scenario.
```

Comment:
>but isn't a high priority scenario. [](http://example.com/codeflow?start=71&length=35)

nit: I'd remove this comment. Just say issue is tracked

Reply:
Done